logging/http_client.go

```diff
@@ -0,0 +1,83 @@
+package logging
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/kr/text"
+)
+
+// HTTPClient is a wrapper around http.Client that logs requests and responses.
+type HTTPClient struct {
+	Wrapped *http.Client
+	Out     io.Writer
+}
+
+// Do performs an HTTP request and logs the request and response.
+func (cli *HTTPClient) Do(req *http.Request) (*http.Response, error) {
+	interpretControlSequences := func(text string) string {
+		text = strings.ReplaceAll(text, `\n`, "\n")
+		text = strings.ReplaceAll(text, `\t`, "\t")
+		return text
+	}
+
```

Comment:
Do we want to also print headers?

Reply:
Could be useful.

Reply:
added headers and path + deffer to write all at once to avoid printing races